### ISSUE TITLE
Can't initialize Selenium::WebDriver::Chrome::Driver with :url

### ISSUE THREAD

**Issue by paddy-hack** [teamcapybara/capybara#2647]
## Meta
Capybara Version: 3.38.0
Driver Information (and browser if relevant): selenium-webdriver 4.8.0, standalone-chrome
 
## Expected Behavior

Tests pass (just like they have been doing when using selenium-webdriver 4.7.1).

## Actual Behavior

Upgrading to selenium-webdriver 4.8.0 makes tests fail with
``` console
     Failure/Error: visit path
     
     ArgumentError:
       Can't initialize Selenium::WebDriver::Chrome::Driver with :url
 ```

## Steps to reproduce

The test suite for our Rails app uses a custom driver for tests that depend on JavaScript getting executed by the browser.   This is configured with

``` ruby
# Capybara tweaks to run JavaScript dependent tests
Capybara.register_driver :remote_selenium_chrome do |app|
  browser_options = ::Selenium::WebDriver::Chrome::Options.new

  Capybara::Selenium::Driver.new app, browser: :chrome,
                                 options: browser_options,
                                 url: "http://standalone-chrome:4444/wd/hub"
end

Capybara.javascript_driver = :remote_selenium_chrome
Capybara.server = :puma, { Silent: true }
Capybara.server_host = `hostname`.strip
```

The `standalone-chrome` "host" runs in a Docker container on the same network as the container that runs the tests.

Our `rspec` tests that need JavaScript are tagged with `js: true`.  That looks a lot like

``` ruby
RSpec.describe "Home Page" do
  scenario "can be visited", js: true do
    visit '/'
  end
end
```

BTW, I have seen #2637.

**Comment by twalpole** [comment]
So, selenium has updated and your custom registration is no longer compatible with it. I'm unclear how this is a Capybara issue?

**Comment by paddy-hack** [comment]
I guess I should have pointed it out but the only place where the URL is passed is in
``` ruby
Capybara::Selenium::Driver.new app, browser: :chrome,
                                 options: browser_options,
                                 url: "http://standalone-chrome:4444/wd/hub"
```

**Comment by twalpole** [comment]
Ok, but that's your code. If selenium no longer accepts that then you need to update your code to pass whatever it accepts now. I don't see how it's a Capybara issue

**Comment by paddy-hack** [comment]
I see it as a Capybara issue because whatever happened under the covers stopped working.  Whatever Capybara did with that `url:` argument stopped working and the only place our code references any selenium-webdriver API is where it instantiates the `browser_options`.

Hmm, peeking under the covers, it seems that all arguments to `Capybara::Selenium::Driver.new` but `app` are merged into an options object that is later processed, removing some special options (not including `url`), and the remaining ones passed in
``` ruby
Selenium::Webdriver.for(options[:browser], processed_options)
```
That isn't exactly obvious from the [Capybara documentation](https://rubydoc.info/github/teamcapybara/capybara/master/Capybara%2FSelenium%2FDriver:initialize) :stuck_out_tongue_winking_eye: 

Digging further still, I found that I needed to pass `browser: :remote` now instead of `browser: :chrome`.  
The commit that triggered this is https://github.com/SeleniumHQ/selenium/commit/08ee2d9fcf952aff3dc9752d2399e9474c356843.

Thanks for pointing me in the right direction to peek and dig.
